src/main.js

```diff
@@ -1,144 +1,175 @@
+    return { dispose: () => cancelIdleCallback(id) }
   }
+}
 
-  if (!atom.inSpecMode()) {
-    makeIdleCallback(linterEslintInstallPeerPackages)
-    makeIdleCallback(linterEslintLoadDependencies)
-    makeIdleCallback(linterEslintStartWorker)
+// // Simplified composite disposable avoids requiring atom core in main
+// //
+const makeCompositeDisposable = () => {
+  const disposables = new Set()
+  return {
+    add: (disposable) => {
+      if (!(disposable.dispose instanceof Function)) {
+        throw new Error('disposable must have dispose method')
+      }
+      disposables.add(disposable)
+    },
+    dispose: () => disposables.forEach(d => d.dispose())
   }
 }
 
+
 module.exports = {
   activate() {
-    getMigrations().map(makeIdleCallback)
-    this.subscriptions = new CompositeDisposable()
+    // const { CompositeDisposable } = require('atom')
+    // this.subscriptions = new CompositeDisposable()
+    this.subscriptions = makeCompositeDisposable()
+
+    // Put dependency loading on Atom's todo-list
+    this.subscriptions.add(makeIdleRunner()(dependencies))
+
+    const {
+      atomConfig,
+      getMigrations,
+      subscribe: configSubscribe
+    } = require('./atom-config')
+
+    // Subscribe to Atom configuration settings
     this.subscriptions.add(...(configSubscribe()))
 
+    // Subscribe to Save event so we can run fixOnSave
     this.subscriptions.add(atom.workspace.observeTextEditors((editor) => {
       editor.onDidSave(async () => {
+        const { hasValidScope } = require('./validate/editor')
+
         const { fixOnSave, scopes } = atomConfig
         if (hasValidScope(editor, scopes) && fixOnSave) {
           await this.fixJob(true)
         }
       })
     }))
 
+    // Subscribe to Debug command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:debug': async () => {
-        loadDeps()
-        const debugString = await debug.report()
-        const notificationOptions = { detail: debugString, dismissable: true }
+        const { report } = require('./debug')
+
+        const debugReport = await report()
+        const notificationOptions = { detail: debugReport, dismissable: true }
         atom.notifications.addInfo('linter-eslint debugging information', notificationOptions)
       }
     }))
 
+    // Subscribe to Fix File command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:fix-file': async () => {
         await this.fixJob()
       }
     }))
 
+    // Add context menu entry for Fix File
     this.subscriptions.add(atom.contextMenu.add({
       'atom-text-editor:not(.mini), .overlayer': [{
         label: 'ESLint Fix',
         command: 'linter-eslint:fix-file',
         shouldDisplay: (evt) => {
           const activeEditor = atom.workspace.getActiveTextEditor()
-          if (!activeEditor) {
-            return false
-          }
-          // Black magic!
+          if (!activeEditor) return false
```

Comment:
I personally absolutely hate inline `if` statements, I thought the ESLint config actually blocked them...

Reply:
(At least JS doesn't allow _postfix_ control structures like some other languages...)

Reply:
No. The config allows 1-liners, but forces braces if you do 2 lines.

I'm not really a huge fan of 1-liners either, very much preferring (in most cases) 2 lines with just an indent and no braces. But I quite strongly dislike using _3_ lines for a statement that is very readable in only 1. And I find braces to be just meaningless noise, much like semicolons. So that was my compromise to comply with the linting rules while reducing useless noise.

Didn't know you had a hatred for them. I guess I'll just quit doing that? 😕

Reply:
There's a reason I sent this as a comment, not requesting changes. This is a style change that can get sorted out in the eventual final merge, but I figured I would mention it since you have started using it more and more.

Reply:
No. It was worth mentioning. If it's gonna get burned anyway, I might as well just get used to it.